### PR TITLE
feature: Add Pub Connections to Pub preview blocks

### PR DESCRIPTION
**Pull request by idreyn**
As seen here:

image

And here in a compact incarnation:

image

Note the new expanded state of the medium and large Pub Preview blocks:

image

Test plan:

- Check out the [storybook fixture](http://localhost:9001/?path=/story/components-pubpreview--default)
- Visit a few different Pages in a few different Communities and verify that any `PubEdge` data shows up as expected in the Connections of relevant Pub previews
- Visit the Page editor and verify that the "Hide Connections" checkbox works.

### CODE REVIEW

File: client/components/Byline/bylineStories.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+					linkToUsers={false}
+					renderUserLabel={(user, index) => `(${index + 1}) ${user.fullName}`}
+				/>
+			</p>
+			<h4>Use ampersand</h4>
+			<p>
+				<Byline contributors={[B, F, G, E]} ampersand />
+			</p>
+		</div>
+	);
```
Comment (3mcd) [inline client/components/Byline/bylineStories.js:83]: The `@storybook/addon-knobs` plugin could work wonders here! Maybe soon we go through and make all of our stories configurable.

File: utils/canonicalUrls.js [hunk 2]
```diff
@@ -42,4 +42,11 @@ export const pubUrl = (community, pub, options = {}) => {
 	);
 };
 
+export const bestPubUrl = ({ pubData, communityData }, options = {}) => {
+	if (communityData) {
+		return pubUrl(communityData, pubData, options);
+	}
+	return pubShortUrl(pubData);
+};
```
Comment (3mcd) [inline utils/canonicalUrls.js:50]: Nice!

**Comment by 3mcd** [comment]
Looks good to me (other than the Storybook failure)!

**Comment by idreyn** [comment]
Storybook knobs look great — just installed and integrated that here.

**Pull request merged**